src/dodal/devices/i24/dual_backlight.py

```diff
@@ -0,0 +1,67 @@
+from ophyd import Component, Device, EpicsSignal, StatusBase
+
+
+class BacklightPositioner(Device):
+    """Device to control the backlight position."""
+
+    # String description of the backlight position e.g. "In", "OAV2"
+    pos_level: EpicsSignal = Component(EpicsSignal, "MP:SELECT")
+
+    zrst: EpicsSignal = Component(EpicsSignal, "MP:SELECT.ZRST")  # Out
+    onst: EpicsSignal = Component(EpicsSignal, "MP:SELECT.ONST")  # In
+    twst: EpicsSignal = Component(EpicsSignal, "MP:SELECT.TWST")
+    thst: EpicsSignal = Component(EpicsSignal, "MP:SELECT.THST")
+    frst: EpicsSignal = Component(EpicsSignal, "MP:SELECT.FRST")
+
+    @property
+    def allowed_backlight_positions(self):
+        return [
+            self.zrst.get(),
+            self.onst.get(),
+            self.twst.get(),
+            self.thst.get(),
+            self.frst.get(),
+        ]
+
+
+class DualBacklight(Device):
+    """
+    Device to trigger the dual backlight on I24.
+    This device is made up by two LEDs:
+        - LED1 is the "backlight", can be moved to 5 different positions.
+        - LED2 is a "frontlight", it does not move, just switches on and off.
+
+    To set the position for LED1:
+        b = DualBacklight(name="backlight)
+        b.pos1.pos_level.set("OAV2")
+
+    To see get the available position values for LED1:
+        b.pos1.alowed_backlight_positions
+
+    Note that the two LED are independently switched on and off. When LED1 is
+    in "Out" position (switched off), LED2 might still be on.
+    """
+
+    OUT = "Out"
+    IN = "In"
+
+    led1: EpicsSignal = Component(EpicsSignal, "-DI-LED-01:TOGGLE")
+    pos1: BacklightPositioner = Component(BacklightPositioner, "-MO-BL-01:")
+
+    led2: EpicsSignal = Component(EpicsSignal, "-DI-LED-02:TOGGLE")
+
+    def set(self, position: str) -> StatusBase:
+        if position not in self.pos1.allowed_backlight_positions:
+            raise ValueError(
+                f"""
+                Unknown requested position for LED1.
+                Allowed values: {self.pos1.allowed_backlight_positions}.
+            """
+            )
+
+        status = self.pos1.pos_level.set(position)
+        if position == self.pos1.zrst.get():
```

Comment:
This implicitly hard-codes the mapping of `ZRST` -> `OUT`, I think. Which is probably unavoidable, but in that case, it might be clearer just to make it *explicit* instead and not try to read zrst/onst etc.